subprojects/docs/src/snippets/tasks/incrementalBuild-customTaskClass/groovy/build.gradle

```diff
@@ -47,6 +47,7 @@ tasks.register('processTemplatesAdHocSkipWhenEmpty') {
         .skipWhenEmpty()
         .withPropertyName('sourceFiles')
         .withPathSensitivity(PathSensitivity.RELATIVE)
+        .ignoreEmptyDirectories()
```

Comment:
Didn't we say that `@SkipWhenEmpty` should imply `@IgnoreEmptyDirectories`? I think it should work the same way in the runtime API.

Reply:
No, we said that `@InputDirectory` implies `@IgnoreEmptyDirectories`. `@SkipWhenEmpty` does not imply `@IgnoreEmptyDirectories`, though it ignores directories with a deprecation warning when the `@SkipWhenEmpty` input property consists only of file trees.

Reply:
Ah, right, okay.